### PR TITLE
 Fix FormRequest.formdata with GET method duplicates same key in querystring (#2919)

### PR DESCRIPTION
**Pull request by iamriel**
This is the fix for issue #2919 This is my first time contributing to an open source project.  This PR is based on gekco's PR [here](https://github.com/scrapy/scrapy/pull/3151).  I followed the instruction in [Submitting Patches](https://doc.scrapy.org/en/master/contributing.html#submitting-patches) for updating an existing PR.  Hopefully I have done it correctly.

### CODE REVIEW

File: scrapy/http/request/form.py [hunk 1]
```diff
@@ -33,7 +33,11 @@ def __init__(self, *args, **kwargs):
                 self.headers.setdefault(b'Content-Type', b'application/x-www-form-urlencoded')
                 self._set_body(querystr)
             else:
-                self._set_url(self.url + ('&' if '?' in self.url else '?') + querystr)
+                query = dict(parse_qsl(urlparse(self.url).query))
+                if query:
+                    query.update(dict(items))
```
Comment (HassanQamar07) [inline scrapy/http/request/form.py]: Sometimes we've to send multiple values for a single key, `dict(items)` will make it impossible and will result in unexpected URL.

`
formdata=[
('a','1'),
('a','2'),
('b','3'),
]
`

**Comment by HassanQamar07** [comment]
@iamriel Can you add another test case which tests the multiple values for same key name
e.g

`formdata=[('a','a')),('a','A')),('b','B')),]`

**Comment by iamriel** [comment]
@HassanQamar07 Thanks for checking this PR, you are right about it, I have updated the implementation and added the test in this [commit](https://github.com/iamriel/scrapy/commit/db7b4250).  Please let me know if there is anything else. Thanks!

**Comment by iamriel** [comment]
Ooop, the build failed, I'll update later today.

**Comment by maramsumanth** [comment]
@iamriel , what happens when there is fragment in url. I think your code fails.

**Comment by iamriel** [comment]
I apologize I haven't had time to attend into this.  I'm closing this as @maramsumanth already had an approved PR for this. Thanks everyone!

**Pull request closed without merging**